jbang/tnb.java

```diff
@@ -0,0 +1,677 @@
+                            ret[i] = (Class<?>) parametrizedClazz.getActualTypeArguments()[i];
+                        }
+                        return ret;
+                    }
+                }
+                clazz = clazz.getSuperclass();
+            }
+        }
+
+        public static Service<?, ?, ?> deployService(String svc, Boolean useOCP) throws Exception {
+            String previousUseOCP = System.getProperty(OpenshiftConfiguration.USE_OPENSHIFT, "false");
+
+            System.setProperty(OpenshiftConfiguration.USE_OPENSHIFT, useOCP.toString());
+            List<Class<? extends Service<?, ?, ?>>> availableSvcs =
+                registeredServices.stream().filter(clazz -> clazz.getSimpleName().contains(svc)).collect(Collectors.toList());
+            if (availableSvcs.size() != 1) {
+                availableSvcs =
+                    availableSvcs.stream().filter(clazz -> clazz.getSimpleName().equals(svc)).collect(Collectors.toList());
+                if (availableSvcs.size() > 1) {
+                    throw new IllegalStateException("Too many services");
+                }
+            }
+            final Class<? extends Service<?, ?, ?>> svcClazz = availableSvcs.get(0);
+            var service = ServiceFactory.create(svcClazz);
+
+            @SuppressWarnings("unchecked") final Class<? extends Account> accountClazz = (Class<? extends Account>) getTypeParams(svcClazz)[0];
+            if (accountClazz != NoAccount.class) {
+                try {
+                    AccountFactory.create(accountClazz);
+                } catch (AssertionFailedError e) {
+                    final List<Field> fields =
+                        ReflectionUtils.findFields(accountClazz,
+                            field -> !ReflectionUtils.isFinal(field) && !ReflectionUtils.isStatic(field) && (field.getType().isPrimitive()
+                                || field.getType().equals(String.class)),
+                            ReflectionUtils.HierarchyTraversalMode.TOP_DOWN);
+                    terminal.writer().println("Missing account credentials for " + accountClazz.getSimpleName());
+                    final Account account = accountClazz.getConstructor().newInstance();
+                    try {
+                        for (Field field : fields) {
+                            final String value = reader.readLine(field.getName() + ": ");
+                            field.setAccessible(true);
+                            FieldUtils.writeField(field, account, value, true);
+                        }
+                    } catch (UserInterruptException ignore) {
+                        terminal.writer().println("\nStopping deployment of " + svcClazz.getSimpleName());
+                        return null;
+                    }
+                    FieldUtils.writeField(service, "account", account, true);
+                }
+            }
+            service.beforeAll(null);
```

Comment:
maybe this should print some log that it is being deployed, otherwise it just "hangs" until its deployed without any log